src/ic-certified-map/src/rbtree.rs

```diff
@@ -189,15 +190,158 @@ impl<K: 'static + AsRef<[u8]>, V: AsHashTree + 'static> Node<K, V> {
     }
 }
 
-/// Implements mutable Leaf-leaning red-black trees as defined in
+enum Position {
+    Left,
+    Middle,
+    Right,
+}
+
+enum IterState<'a, K, V> {
```

Comment:
Could you please add invariants for the state?
The last entry in `Deep`:
(p, Left) => the left subtree of p is visited
(p, Middle) => the left subtree of p and p itself are visited
(p, Right) => p and its subtrees are visited.

Reply:
Good point, documented the invariants for Iter.